actors/common/src/main/java/org/sunbird/learner/actors/data/security/manager/DataSecurityBackgroundActor.java

```diff
@@ -0,0 +1,184 @@
+package org.sunbird.learner.actors.data.security.manager;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.sunbird.actor.core.BaseActor;
+import org.sunbird.actor.router.ActorConfig;
+import org.sunbird.cassandra.CassandraOperation;
+import org.sunbird.common.exception.ProjectCommonException;
+import org.sunbird.common.models.response.Response;
+import org.sunbird.common.models.util.ActorOperations;
+import org.sunbird.common.models.util.JsonKey;
+import org.sunbird.common.models.util.LoggerEnum;
+import org.sunbird.common.models.util.ProjectLogger;
+import org.sunbird.common.models.util.ProjectUtil;
+import org.sunbird.common.request.Request;
+import org.sunbird.common.responsecode.ResponseCode;
+import org.sunbird.helper.ServiceFactory;
+import org.sunbird.learner.util.UserUtility;
+import org.sunbird.learner.util.Util;
+
+/** Background sync of data between Cassandra and Elastic Search. */
+@ActorConfig(
+  tasks = {},
+  asyncTasks = {"backgroundEncryption", "backgroundDecryption"}
+)
+public class DataSecurityBackgroundActor extends BaseActor {
+
+  private CassandraOperation cassandraOperation = ServiceFactory.getInstance();
+  private Util.DbInfo usrDbInfo = Util.dbInfoMap.get(JsonKey.USER_DB);
+  private Util.DbInfo addrDbInfo = Util.dbInfoMap.get(JsonKey.ADDRESS_DB);
+
+  @Override
+  public void onReceive(Request request) throws Throwable {
+    String operation = request.getOperation();
+    switch (operation) {
+      case "backgroundEncryption":
+        backgroundEncrypt(request);
+        break;
+      case "backgroundDecryption":
+        backgroundDecrypt(request);
+        break;
+      default:
+        onReceiveUnsupportedOperation("EsSyncBackgroundActor");
+        break;
+    }
+  }
+
+  private void backgroundEncrypt(Request request) {
+    List<Map<String, Object>> userDetails = getUserDetails(request);
+    commonEncryptionDecryption(userDetails, true);
+  }
+
+  private void backgroundDecrypt(Request request) {
+    List<Map<String, Object>> userDetails = getUserDetails(request);
+    commonEncryptionDecryption(userDetails, false);
+  }
+
+  private void commonEncryptionDecryption(
+      List<Map<String, Object>> userDetails, boolean isEncrypted) {
+    List<String> userIdsListToSync = new ArrayList<>();
+    int i = 0;
+    int maxPhoneNoSizeAllowed =
+        Integer.parseInt(ProjectUtil.getConfigValue(JsonKey.SUNBIRD_USER_MAX_PHONE_LENGTH).trim());
```

Comment:
Don't use length. Rather use regex for phone number. Check with Manzarul already somewhere in code we could be using it.